### ISSUE TITLE
num/dualcmplx: Abs() should return Number type as num/dual does

### ISSUE THREAD

**Issue by nisfornature** [gonum/gonum#1093]
### What are you trying to do?

Simplify and standardize the return types of  functions that work with special numbers (e.g. dual complex numbers, grassmann numbers).

Namely, num/dualcmplx/dual.go Abs() takes a Number type, however, supplies a float64 .

num/dual/dual.go Abs() presently returns the magnitude of the supplied number type wrapped as the dual Number struct as so:

```
// Abs returns the absolute value of d.
func Abs(d Number) Number {
        if !math.Signbit(d.Real) {
                return d
        }
        return Scale(-1, d)
}
```

The absolute value of a complex number always yields a real  number. In Golang, taking the absolute value of a complex128 number yields a float64. Unlike the dual real in num/dual/dual.go, taking the absolute value of a dual complex does not preserve the dual type.

I feel the typing should be preserved.

### What did you try?
```
// Abs returns the absolute value of d.
func Abs(d Number) Number { 
        return Number{
                Real: complex(cmplx.Abs(d.Real), 0),
                Dual: d.Dual,
        }
}
```

Such that:
`Number{Real: cmplx.NaN(), Dual: 0+3i}` returns, `Number{Real: complex(math.NaN(),0), Dual: 0+3i}}` 
and
`{d: Number{Real: negZeroCmplx, Dual: 1.175495e-38i}` returns, `Number{Real: 0+0i, Dual: 1.175495e-38i}}` .

### How does Gonum not allow you to achieve your goal?
When working with calls to num/dualcmplx/dual.go Abs(), I need to be mindful of float64 values.

### What version of Go and Gonum are you using?
Go: v1.13
Gonum: 16f41441268f3dab0269c341936d645ec725d07d

### Is this feature absent from the current master?
Yes.

### Are you able to help contribute the feature?
Yes.

**Comment by kortschak** [comment]
I don't think this is the correct thing to do. Note that the num/dual and num/cmplxdual are applied to very different domains, the latter being primarily used for deformations in the plane while the former is primarily for autodiff. Because of this they have different primitives; the implementations for dual and (anti-commutative) complex dual numbers reflect the original mathematical work and don't necessarily need to be consistent with each other. If we introduced a commutative complex dual implementation, then it would likely agree with the real dual implementation.
